### ISSUE TITLE
Firmware Update never completes

### ISSUE THREAD

**Issue by dannytsang** [Dygmalab/Bazecor#402]
**Describe the bug**
The firmware update starts and gets to **Firmware update!** Solid as a rock! 💪 but Bazecor hangs.

**To Reproduce**
1. Go to Firmware Update
2. Click on the bagillion dialogues to start the firmware upgrade
3. Follow the instructions to start the firmware.
4. See this screen:

**Expected behavior**
The updater to complete and let me get back to using Bazecor.

**Screenshots**
![image](https://user-images.githubusercontent.com/567982/215882730-bb1c7edf-fb3f-470b-9e92-a2e3236fbe35.png)

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Bazecor Version: 1.0

**Additional context**
At the screenshot above stage, it's the neuron is flashing blue. Looking at the screenshot I assume it's supposed to turn solid but never does. I close Bazecor and it detects the neuron is in bootloader mode but it can never start the firmware upgrade process even holding the escape key and I have to unplug and replug it back in.

**Comment by amazingmaze** [comment]
Had the same error. Had to unplug the keyword which wiped all settings. Another note,it should say "Firmware updated!" and maybe remove the loading bar. Not good UX atm :)

**Comment by HybridEidolon** [comment]
Same issue. Claims it's on firmware 1.0.9beta (I am missing the new light modes that were in the beta firmwares before 1.0), attempting to reflash causes this issue.

**Comment by HybridEidolon** [comment]
Actually, 1.0.9beta must be on older code than 0.3.5 released last year. I think y'all have multiple firmware releases out of sync with Bazecor and something is broken.

I tripped the flashing bug flashing 0.3.5, but 0.3.5 is far more responsive and has more features than the 1.0.9beta Bazecor 1.0 recommends installing. Please look into this; the average user is going to install 1.0.9beta as recommended and have a degraded user experience out of the box.

**Comment by AlexDygma** [comment]
Hey @HybridEidolon! We are in fact addressing the 0.3.5 issue mismatch right now, 0.3.5 is the way to go for almost everyone, so we are going to add the proper channel to use the version on the latest Bazecor version, only the people that want superkeys will be using the beta firmware, which is still to get out of beta due to the development of the Defy keyboard.

The flashing issues should not be present on the newer version of Bazecor and in fact, we are rebuilding the whole flashing interface to add the Defy steps, remote firmware repositories, and updated packages which should push up stability.

Feel free to reopen this issue if, after the next update, the FW update fails for you! 💪

**Comment by sfavazza** [comment]
Hello!

Just got my Defy keyboard and getting issues to update the neuron. The bazecore detects version `v0.0.6` as in the screenshot below:
![image](https://github.com/Dygmalab/Bazecor/assets/36377461/35f39097-4e14-4430-8354-38a9dc32cb59)

Though when trying to update it hangs at the neuron update step:
![image](https://github.com/Dygmalab/Bazecor/assets/36377461/51a8cc72-5402-4618-a047-ee65f0b431e0)

I tried to repeat the process starting with the lowest available version offered (`v1.0.0`) but I got the same result.
Restoring the backup works though, so maybe I am not in an unstable state. 

Is there any diagnostic step I can perform to further investigate on the issue?

Further info:
- I counted up to 8 processes running bazecor software
- the CPU works close to 100%

**Comment by xaionaro** [comment]
Having the same issue as @sfavazza .
Mine is happening on Linux, and CPU is consumed by a lot of calls of "lsblk" by Bazecor. Tried different cables and USB ports and firmware target versions -- nothing helps.
```
           |-alacritty(52035)-+-bash(52036)---Bazecor(47020)-+-lsblk(7544)
           |                  |                              |-lsblk(7545)
           |                  |                              |-lsblk(7546)
           |                  |                              |-lsblk(7547)
           |                  |                              |-lsblk(7548)
           |                  |                              |-lsblk(7549)
           |                  |                              |-lsblk(7550)
           |                  |                              |-lsblk(7551)
           |                  |                              |-lsblk(7552)
           |                  |                              |-lsblk(7553)
           |                  |                              |-lsblk(7554)
           |                  |                              |-lsblk(7555)
           |                  |                              |-lsblk(7556)
           |                  |                              |-lsblk(7557)
           |                  |                              |-lsblk(7564)
           |                  |                              |-lsblk(7568)
           |                  |                              |-lsblk(7570)
           |                  |                              |-lsblk(7571)
           |                  |                              |-lsblk(7572)
           |                  |                              |-lsblk(7573)
           |                  |                              |-lsblk(7574)
           |                  |                              |-lsblk(7575)
           |                  |                              |-lsblk(7576)
           |                  |                              |-lsblk(7577)
           |                  |                              |-lsblk(7578)
           |                  |                              |-lsblk(7579)
           |                  |                              |-lsblk(7580)
           |                  |                              |-lsblk(7581)
           |                  |                              |-lsblk(7582)
           |                  |                              |-lsblk(7583)
           |                  |                              |-lsblk(7584)
           |                  |                              |-lsblk(7585)
           |                  |                              |-lsblk(7586)
           |                  |                              |-lsblk(7587)
           |                  |                              |-lsblk(7588)
           |                  |                              |-lsblk(7589)
           |                  |                              |-lsblk(7590)
           |                  |                              |-lsblk(7591)
           |                  |                              |-lsblk(7592)
           |                  |                              |-lsblk(7593)
           |                  |                              |-lsblk(7594)
           |                  |                              |-lsblk(7595)
           |                  |                              |-lsblk(7596)
           |                  |                              |-lsblk(7597)
           |                  |                              |-lsblk(7598)
           |                  |                              |-lsblk(7599)
           |                  |                              |-lsblk(7600)
           |                  |                              |-lsblk(7601)
           |                  |                              |-lsblk(7602)
           |                  |                              |-lsblk(7603)
           |                  |                              |-lsblk(7604)
           |                  |                              |-lsblk(7605)
           |                  |                              |-lsblk(7606)
           |                  |                              |-lsblk(7607)
           |                  |                              |-lsblk(7608)
           |                  |                              |-lsblk(7609)
           |                  |                              |-lsblk(7610)
           |                  |                              |-lsblk(7611)
           |                  |                              |-lsblk(7612)
           |                  |                              |-lsblk(7613)
           |                  |                              |-lsblk(7614)
           |                  |                              |-lsblk(7615)
           |                  |                              |-lsblk(7616)
           |                  |                              |-lsblk(7617)
           |                  |                              |-lsblk(7618)
           |                  |                              |-lsblk(7619)
           |                  |                              |-lsblk(7620)
           |                  |                              |-lsblk(7621)
           |                  |                              |-lsblk(7622)
           |                  |                              |-lsblk(7623)
           |                  |                              |-lsblk(7624)
           |                  |                              |-lsblk(7625)
           |                  |                              |-lsblk(7626)
           |                  |                              |-lsblk(7627)
           |                  |                              |-lsblk(7628)
           |                  |                              |-lsblk(7629)
           |                  |                              |-lsblk(7630)
           |                  |                              |-lsblk(7631)
           |                  |                              |-lsblk(7632)
           |                  |                              |-lsblk(7633)
           |                  |                              |-lsblk(7634)
           |                  |                              |-lsblk(7635)
           |                  |                              |-lsblk(7636)
           |                  |                              |-lsblk(7637)
           |                  |                              |-lsblk(7638)
           |                  |                              |-lsblk(7639)
           |                  |                              |-lsblk(7640)
           |                  |                              |-lsblk(7641)
           |                  |                              |-lsblk(7642)
           |                  |                              |-lsblk(7643)
           |                  |                              |-lsblk(7644)
           |                  |                              |-lsblk(7645)
           |                  |                              |-lsblk(7646)
           |                  |                              |-lsblk(7647)
           |                  |                              |-lsblk(7648)
           |                  |                              |-lsblk(7649)
           |                  |                              |-lsblk(7650)
           |                  |                              |-lsblk(7651)
           |                  |                              |-lsblk(7652)
           |                  |                              |-lsblk(7653)
           |                  |                              |-lsblk(7654)
           |                  |                              |-lsblk(7655)
           |                  |                              |-lsblk(7656)
           |                  |                              |-lsblk(7657)
           |                  |                              |-lsblk(7658)
           |                  |                              |-lsblk(7659)
           |                  |                              |-lsblk(7660)
           |                  |                              |-lsblk(7661)
           |                  |                              |-lsblk(7662)
           |                  |                              |-{Bazecor}(47026)
           |                  |                              |-Bazecor(47028)---Bazecor(47057)-+-Bazecor(47087)
           |                  |                              |                                 |-{Bazecor}(47088)
           |                  |                              |                                 |-{Bazecor}(47089)
           |                  |                              |                                 |-{Bazecor}(47090)
           |                  |                              |                                 |-{Bazecor}(47091)
           |                  |                              |                                 |-{Bazecor}(47092)
           |                  |                              |                                 |-{Bazecor}(47093)
           |                  |                              |                                 |-{Bazecor}(47094)
           |                  |                              |                                 |-{Bazecor}(47112)
           |                  |                              |                                 |-{Bazecor}(47113)
           |                  |                              |                                 |-{Bazecor}(47114)
           |                  |                              |                                 |-{Bazecor}(47115)
           |                  |                              |                                 |-{Bazecor}(47116)
           |                  |                              |                                 |-{Bazecor}(47117)
           |                  |                              |                                 |-{Bazecor}(47118)
           |                  |                              |                                 |-{Bazecor}(47119)
           |                  |                              |                                 |-{Bazecor}(47120)
           |                  |                              |                                 |-{Bazecor}(47121)
           |                  |                              |                                 |-{Bazecor}(47122)
           |                  |                              |                                 |-{Bazecor}(47123)
           |                  |                              |                                 |-{Bazecor}(47124)
           |                  |                              |                                 |-{Bazecor}(47125)
           |                  |                              |                                 |-{Bazecor}(47126)
           |                  |                              |                                 |-{Bazecor}(47127)
           |                  |                              |                                 |-{Bazecor}(47128)
           |                  |                              |                                 |-{Bazecor}(47129)
           |                  |                              |                                 |-{Bazecor}(47130)
           |                  |                              |                                 |-{Bazecor}(47131)
           |                  |                              |                                 |-{Bazecor}(47132)
           |                  |                              |                                 |-{Bazecor}(47133)
           |                  |                              |                                 |-{Bazecor}(47134)
           |                  |                              |                                 |-{Bazecor}(47135)
           |                  |                              |                                 `-{Bazecor}(48087)
           |                  |                              |-Bazecor(47029)---Bazecor(47031)
           |                  |                              |-{Bazecor}(47032)
           |                  |                              |-{Bazecor}(47033)
           |                  |                              |-{Bazecor}(47034)
           |                  |                              |-{Bazecor}(47035)
           |                  |                              |-{Bazecor}(47036)
           |                  |                              |-{Bazecor}(47037)
           |                  |                              |-{Bazecor}(47038)
           |                  |                              |-{Bazecor}(47039)
           |                  |                              |-{Bazecor}(47040)
           |                  |                              |-{Bazecor}(47041)
           |                  |                              |-{Bazecor}(47042)
           |                  |                              |-{Bazecor}(47043)
           |                  |                              |-{Bazecor}(47044)
           |                  |                              |-{Bazecor}(47045)
           |                  |                              |-{Bazecor}(47046)
           |                  |                              |-{Bazecor}(47047)
           |                  |                              |-{Bazecor}(47048)
           |                  |                              |-{Bazecor}(47049)
           |                  |                              |-{Bazecor}(47051)
           |                  |                              |-{Bazecor}(47052)
           |                  |                              |-{Bazecor}(47053)
           |                  |                              |-{Bazecor}(47054)
           |                  |                              |-{Bazecor}(47055)
           |                  |                              |-{Bazecor}(47056)
           |                  |                              |-{Bazecor}(47058)
           |                  |                              |-{Bazecor}(47059)
           |                  |                              |-{Bazecor}(47060)
           |                  |                              |-Bazecor(47063)-+-{Bazecor}(47069)
           |                  |                              |                |-{Bazecor}(47070)
           |                  |                              |                |-{Bazecor}(47071)
           |                  |                              |                |-{Bazecor}(47072)
           |                  |                              |                |-{Bazecor}(47073)
           |                  |                              |                |-{Bazecor}(47074)
           |                  |                              |                `-{Bazecor}(47075)
           |                  |                              |-{Bazecor}(47064)
           |                  |                              |-{Bazecor}(47066)
           |                  |                              |-Bazecor(47067)-+-{Bazecor}(47076)
           |                  |                              |                |-{Bazecor}(47077)
           |                  |                              |                |-{Bazecor}(47078)
           |                  |                              |                |-{Bazecor}(47079)
           |                  |                              |                |-{Bazecor}(47080)
           |                  |                              |                |-{Bazecor}(47085)
           |                  |                              |                |-{Bazecor}(47086)
           |                  |                              |                |-{Bazecor}(47095)
           |                  |                              |                |-{Bazecor}(47096)
           |                  |                              |                |-{Bazecor}(47097)
           |                  |                              |                |-{Bazecor}(47098)
           |                  |                              |                |-{Bazecor}(47103)
           |                  |                              |                |-{Bazecor}(47104)
           |                  |                              |                |-{Bazecor}(47105)
           |                  |                              |                |-{Bazecor}(47106)
           |                  |                              |                |-{Bazecor}(47136)
           |                  |                              |                |-{Bazecor}(47137)
           |                  |                              |                |-{Bazecor}(47138)
           |                  |                              |                |-{Bazecor}(47139)
           |                  |                              |                |-{Bazecor}(47141)
           |                  |                              |                |-{Bazecor}(47142)
           |                  |                              |                |-{Bazecor}(47143)
           |                  |                              |                |-{Bazecor}(47144)
           |                  |                              |                |-{Bazecor}(47146)
           |                  |                              |                |-{Bazecor}(47147)
           |                  |                              |                |-{Bazecor}(47148)
           |                  |                              |                |-{Bazecor}(47149)
           |                  |                              |                |-{Bazecor}(47150)
           |                  |                              |                |-{Bazecor}(47151)
           |                  |                              |                |-{Bazecor}(47152)
           |                  |                              |                |-{Bazecor}(47153)
           |                  |                              |                |-{Bazecor}(47154)
           |                  |                              |                |-{Bazecor}(47155)
           |                  |                              |                |-{Bazecor}(48085)
           |                  |                              |                `-{Bazecor}(48086)
           |                  |                              |-{Bazecor}(47068)
           |                  |                              |-{Bazecor}(47107)
           |                  |                              |-{Bazecor}(47108)
           |                  |                              |-{Bazecor}(48184)
           |                  |                              |-{Bazecor}(48185)
           |                  |                              |-{Bazecor}(48186)
           |                  |                              `-{Bazecor}(48187)
```

**Comment by sfavazza** [comment]
Forgot to mention that I also got the issue on Linux as well. Starting the app from terminal I got many messages about `drive info`:
![image](https://github.com/Dygmalab/Bazecor/assets/36377461/f9f5b3d8-7769-4d63-bc4b-8985b1de498c)


I also tried to follow the [official guide](https://support.dygma.com/hc/en-us/articles/15270931302813-How-to-restore-my-Defy-to-factory-settings) to restore the factory settings, but the neuron flashing hangs at the point depicted in the screenshot below. The messages appearing in the terminal are the same as in the screenshot above. Something seems wrong with the neuron itself...
![image](https://github.com/Dygmalab/Bazecor/assets/36377461/1ca61f39-8f1c-4bf1-b0c6-7317d9a50081)

I also tried to restore the firmware starting the neuron in bootloader mode (following this [guide](https://support.dygma.com/hc/en-us/articles/15268333842333-I-can-t-flash-the-firmware-of-my-Defy)) and I got a `Communication timeout` message. Trying to update the firmware resulted in the `error.platform.FocusAPIRead` error message (see screenshots below):

![image](https://github.com/Dygmalab/Bazecor/assets/36377461/f358a53b-0f9e-4b6e-a5ee-5a73b4177c48)


![image](https://github.com/Dygmalab/Bazecor/assets/36377461/92617944-96b3-4370-89dc-474d55ca462f)

**Comment by xaionaro** [comment]
Since this issue is closed, opening a new one: #643 .
